### PR TITLE
Add unit test ci job #3

### PR DESCRIPTION
**Pull request by rjtch**
added pipeline for tests

### CODE REVIEW

**Comment by davidB** [comment]
@rjtch I prefer to have as few as possible workflow files (eg 1 `ci.yml` file) with all the job/steps to the ci, especially if they
- used the same triggers
- used the same setup (aka the setup of the developer, who should be able to run the same on local)
- are not shared/reused by other repo,...

I also try to limit as much as possible setup that could not be reproduced by the developer locally (=> why I prefer mise or devcontainer over using a tool that configures toolchain on for ci like `setup-rust-toolchain`, `install-action`, including `cargo-make`(made by me :-D)). FYI, `mise` could be installed in ci with curl (in this case, there is no cache).

Creating concurrent jobs (with matrix) as done for lint and tests (in this PR) looks friendly in terms of visual report and execution time (in parallel). But I prefer sequential with fewer jobs because:
- fewer resource usage (fewer machines to up, fewer times to run the setup, less download/upload of cache, and less conflict on shared cache), we can stop earlier (on the first failure.
- more straightforward to "matrix" for multiple environments (windows, linux-x86_64, linux-arm64,...)

I'm sure in conventional commit we should use prefix `ci:` (or `build`) instead of `feat(ci):`.

**Comment by davidB** [comment]
```suggestion
jobs:
  tests:
    runs-on: ${{ matrix.os }}
    strategy:
      fail-fast: false
      matrix:
        os: [ubuntu-latest]
    env:
      CARGO_TERM_COLOR: always
      RUST_BACKTRACE: full
    steps:
      - uses: actions/checkout@v4
        with:
          fetch-depth: 0
          submodules: "true"
      # - uses: jdx/mise-action@v2
      - name: install mise until mise-action is allowed
        run: |
          curl https://mise.jdx.dev/install.sh | sh
          mise install
      - name: Rust Cache
        uses: actions/cache@v4
        continue-on-error: false
        with:
          path: |
            ~/.cargo/bin
            ~/.cargo/registry
            ~/.cargo/git/db/
          key: ${{ runner.os }}-sdk-rust-${{ hashFiles('**/Cargo.toml') }} # no Cargo.lock in this repo
          restore-keys: |
            ${{ runner.os }}-sdk-rust-
      - run: make check
      - run: make generate
      - run: make check_no_uncommitted_changes_on_sdk
      - run: make test
```

**Comment by davidB** [comment]
reminder: `Cargo.lock` are not committed for lib (only for application). So don't expect to use it in any action, command launched by ci (cache, call to make,...)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          submodules: "true"
+      # - uses: jdx/mise-action@v2
+      - name: install mise until mise-action is allowed
+        uses: taiki-e/install-action@v2
+        with:
+          tool: mise, cargo-nextest, cargo-hack
```
Comment (afrittoli) [inline .github/workflows/ci.yml:29]: Please let me know which action you need exactly and I will add it to the configuration.

**Review by afrittoli — APPROVED** [review]

**Comment by afrittoli** [comment]
Thanks for all the updates!

**Pull request merged**